OpenRA.Game/Map/MapDirectoryTracker.cs

```diff
@@ -0,0 +1,121 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2022 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using OpenRA.FileSystem;
+
+namespace OpenRA
+{
+	public class MapDirectoryTracker
+	{
+		readonly FileSystemWatcher watcher;
```

Comment:
Because `FileSystemWatcher ` is `IDisposable`, we need to ensure that logic is plumbed through, so we don't have file watchers living forever once we stop caring about them.

- Change the class to `public sealed class MapDirectoryTracker : IDisposable`
- Implement `public void Dispose` with a body of `watcher.Dispose();`
- In `MapCache`, update the `Dispose` method to also loop over and dispose every tracker.

Reply:
done